src/Storages/System/StorageSystemTables.cpp

```diff
@@ -53,6 +53,9 @@ StorageSystemTables::StorageSystemTables(const StorageID & table_id_)
         {"lifetime_bytes", std::make_shared<DataTypeNullable>(std::make_shared<DataTypeUInt64>())},
         {"comment", std::make_shared<DataTypeString>()},
         {"has_own_data", std::make_shared<DataTypeUInt8>()},
+        {"parts", std::make_shared<DataTypeUInt64>()},
+        {"active_parts", std::make_shared<DataTypeUInt64>()},
+        {"total_marks", std:make_shared<DataTypeUInt64>()},
```

Comment:
But it is not enough to just add columns - you need to fill them with values, otherwise they will always be empty. See `TablesBlockSource::generate`. Also need to add a test, see `tests/queries/0_stateless/` and https://clickhouse.com/docs/en/development/tests/.

Reply:
Noted,
So performing insertion as in from line 182 to 274? and adding the columns to the end of column list?? Gotcha

Reply:
not necessarily to the end, but, yes, should be filled like that